coderd/workspaceagents.go

```diff
@@ -1365,7 +1366,9 @@ func (api *API) workspaceAgentReportStats(rw http.ResponseWriter, r *http.Reques
 		}
 		return nil
 	})
-	if req.SessionCount() > 0 {
+	// NOTE(Cian): we previously checked if SessionCount() > 0; however this does not handle the
+	// case where a user is only accessing their workspace via a coder_app (e.g. code-server).
+	if req.RxBytes > 0 || req.RxPackets > 0 {
```

Comment:
This is not fully indicative of _user_ activity, it will also include communication between coderd and the agent. Perhaps we should look at the data to deduce if this will be a problem or not?

We could also use the workspace app stats to deduce if a session is actually active, i.e. either `req.SessionCount` or app stats (e.g. `workspace_app_stats` table).

Reply:
> This is not fully indicative of user activity, it will also include communication between coderd and the agent. Perhaps we should look at the data to deduce if this will be a problem or not?

I've checked the agent stats payloads from coderd logs, and I've found that when I close all connections to the workspace these fields fairly consistently remain at 0. 

> We could also use the workspace app stats to deduce if a session is actually active, i.e. either req.SessionCount or app stats (e.g. workspace_app_stats table).

AFAICT would need to check both `SessionCount` and `workspace_app_stats`, as those are two different data points entirely (SessionCount appears to only relate to agent SSH connections, while app stats only relates to coder_apps). Checking RX bytes has the nice property of being very simple.

Reply:
Yeah, checking both is what I was going for. For example, my guess is that viewing the workspace in the dashboard (which requests current ports from agent) should produce some non-zero values here, but I could be wrong.

It's good if it's primarily zero when not in use, though. I suppose that makes this an acceptable solution for now, but I do think we should greatly enhance how we track this information. It'd be nice if we could send some "connection headers" so that different types of connections can be identified by the agent.

Reply:
> Yeah, checking both is what I was going for. For example, my guess is that viewing the workspace in the dashboard (which requests current ports from agent) should produce some non-zero values here, but I could be wrong.

You are correct, that does produce non-zero `RxBytes` activity.

Reply:
I think it shows how many ports are open, even when not clicked. So that would have to disappear. OTOH, I don’t see why the port request should trigger “last use” at all. It’s basically just information about the workspace, and the implementation seems to me like a simplification so that we don’t need to report ports real-time to coderd and store in the DB.

Reply:
> OTOH, I don’t see why the port request should trigger “last use” at all.

If someone has to click a button to see it, it becomes a signal of last use.